src/testing/cluster/state_checker.zig

```diff
@@ -179,7 +179,9 @@ pub fn StateCheckerType(comptime Client: type, comptime Replica: type) type {
                     if (client.id == header_b.?.client) break client;
                 } else unreachable;
 
-                if (client.request_inflight == null) {
+                if (client.register_inflight == null and
+                    client.request_inflight == null)
```

Comment:
Would you want to merge this check with the extraction of the `const request` message down below?